scripts/cat/cats.py

```diff
@@ -819,79 +824,67 @@ def add_to_clan(self) -> list:
         for child_id in children:
             child = Cat.all_cats[child_id]
             if (
-                child.outside
-                and not child.exiled
-                and not child.dead
-                and child.moons < 12
+                    child.outside
+                    and not child.exiled
+                    and not child.dead
+                    and child.moons < 12
             ):
                 child.add_to_clan()
                 History.add_beginning(child)
                 ids.append(child_id)
 
         return ids
 
-    def status_change(self, new_status, resort=False):
+    def status_change(self, new_status: StatusEnum | str, resort=False):
         """Changes the status of a cat. Additional functions are needed if you want to make a cat a leader or deputy.
-        new_status = The new status of a cat. Can be 'apprentice', 'medicine cat apprentice', 'warrior'
-                    'medicine cat', 'elder'.
+        new_status = The new status of a cat.
         resort = If sorting type is 'rank', and resort is True, it will resort the cat list. This should
                 only be true for non-timeskip status changes."""
+
+        # auto-converting string
+        if isinstance(new_status, str):
+            new_status = StatusEnum(new_status)
+
         old_status = self.status
         self.status = new_status
         self.name.status = new_status
 
+        if new_status.is_outside_clan():
+            return
```

Comment:
Why return early? Doesn't this skip the updating of the `all_cats` dictionary?

Reply:
Good point, I have no idea why this is here, evicting it immediately.